### PR TITLE
TST/CLN: series.duplicated; parametrisation; fix warning

### PR DESCRIPTION
**Pull request by h-vetinari**
Splitting up #21645

Added tests for `duplicated`, parametrized two tests for `drop_duplicates`, fixed a warning from #21614.

### CODE REVIEW

File: pandas/tests/series/test_analytics.py [hunk 1]
```diff
@@ -956,94 +956,56 @@ def test_unique(self):
...
-        assert_series_equal(tc2.duplicated(keep=False), expected)
-        assert_series_equal(tc2.drop_duplicates(keep=False), tc2[~expected])
-        sc = tc2.copy()
-        sc.drop_duplicates(keep=False, inplace=True)
-        assert_series_equal(sc, tc2[~expected])
+            ('first', Series([False, False, False, False, True, True, False])),
+            ('last', Series([False, True, True, False, False, False, False])),
+            (False, Series([False, True, True, False, True, True, False]))
+        ])
+    @pytest.mark.parametrize('npdtype', ['int_', 'uint', 'float_', 'unicode_'])
```
Comment (jreback) [inline pandas/tests/series/test_analytics.py]: can you use the fixtures defined in pandas/conftest for the dtypes cc @gfyoung

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/series/test_analytics.py (hunk 1 above)
Comment (h-vetinari) [reply pandas/tests/series/test_analytics.py]: @jreback 
I tried
```
@pytest.mark.parametrize('npdtype', [any_int_dtype, float_dtype, string_dtype])
```
but pytest can't feed fixtures into parametrisation, see https://github.com/pytest-dev/pytest/issues/349.

Not sure how to do this - separate tests?

Comment (jreback) [reply pandas/tests/series/test_analytics.py]: you simply create a composite fixture

**Comment by jreback** [comment]
needs a rebase

File: pandas/conftest.py [hunk 2]
```diff
@@ -338,6 +337,13 @@ def any_int_dtype(request):
     return request.param
 
 
+FLOAT_DTYPES = [float, "float32", "float64"]
+COMPLEX_DTYPES = [complex, "complex64", "complex128"]
+STRING_DTYPES = [str, 'str', 'U']
+ALL_REAL_DTYPES = FLOAT_DTYPES + ALL_INT_DTYPES
+ALL_NUMPY_DTYPES = ALL_REAL_DTYPES + STRING_DTYPES + COMPLEX_DTYPES
+
+
```
Comment (gfyoung) [inline pandas/conftest.py]: Make sure that these lists aren't duplicated in the file.

Comment (gfyoung) [reply pandas/conftest.py]: One block of all-caps definitions and use it everywhere in the file.  For example, search for : 

`float, "float32", "float64"`

in your version of `conftest.py` .  You'll see it exists twice in the file.

Comment (jreback) [reply pandas/conftest.py]: pls put these with the other definitions

File: pandas/tests/series/test_duplicates.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        _, err = capsys.readouterr()
+        assert len(err) == 0
+
+    @pytest.mark.parametrize(
+        'keep, expected',
+        [
+            ('first', Series([False, False, False, False, True, True, False])),
+            ('last', Series([False, True, True, False, False, False, False])),
+            (False, Series([False, True, True, False, True, True, False]))
+        ])
```
Comment (jreback) [inline pandas/tests/series/test_duplicates.py]: either use 2 classes for test definitions or don’t use them (preferred)

Comment (jreback) [reply pandas/tests/series/test_duplicates.py]: same for frame

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/series/test_duplicates.py (hunk 3 above)
Comment (h-vetinari) [reply pandas/tests/series/test_duplicates.py]: Classes of tests? I just wrote them in the same manner as in `test_analytics` and then transferred them here. You want me to get rid of `TestSeriesDuplicates` and do class-less tests - do I get that correctly?

Comment (jreback) [reply pandas/tests/series/test_duplicates.py]: yes that is the old format is much less readable

we only use classes if they r logical separations

File: pandas/tests/series/test_duplicates.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
+# coding=utf-8
+# pylint: disable-msg=E1101,W0612
```
Comment (jreback) [inline pandas/tests/series/test_duplicates.py]: what are you disabling?

File: pandas/tests/series/test_duplicates.py [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+import pytest
+
+import numpy as np
+import pandas as pd
```
Comment (jreback) [inline pandas/tests/series/test_duplicates.py]: don’t import pd, directly import instead

File: pandas/tests/series/test_duplicates.py [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+from pandas import Series
+
+from pandas.util.testing import assert_series_equal
+import pandas.util.testing as tm
```
Comment (jreback) [inline pandas/tests/series/test_duplicates.py]: use tm; don’t import assert_series_equal

**Review by jreback — CHANGES_REQUESTED** [review]
for new modules we like to clean things up to conform

**Comment by h-vetinari** [comment]
@jreback All just left-overs from the previous module. Hopefully converging on a clean module soon... ;-)

**Comment by jreback** [comment]
@h-vetinari absolutely - is just easier to fix when exracting things and make nice and clean

File: pandas/conftest.py [hunk 7]
```diff
@@ -248,7 +248,19 @@ def tz_aware_fixture(request):
     return request.param
 
 
-@pytest.fixture(params=[str, 'str', 'U'])
+UNSIGNED_INT_DTYPES = ["uint8", "uint16", "uint32", "uint64"]
+SIGNED_INT_DTYPES = [int, "int8", "int16", "int32", "int64"]
```
Comment (jreback) [inline pandas/conftest.py]: @gfyoung this is ok? it follows our existing pattern?

File: pandas/tests/series/test_duplicates.py [hunk 8]
```diff
@@ -0,0 +1,141 @@
+# coding=utf-8
+
+import pytest
+
+import numpy as np
```
Comment (jreback) [inline pandas/tests/series/test_duplicates.py]: you can remove space between pandas imports

**Review by jreback — CHANGES_REQUESTED** [review]
can u do the same for frame? (unique and duplicates)

File: pandas/conftest.py (hunk 7 above)
Comment (h-vetinari) [reply pandas/conftest.py]: @jreback @gfyoung I only added `COMPLEX_DTYPES`, `STRING_DTYPES` and `ALL_NUMPY_DTYPES` - the rest was just reordering

**Comment by h-vetinari** [comment]
> can u do the same for frame? (unique and duplicates)
@jreback I did the same for frame in #21898

**Comment by h-vetinari** [comment]
@jreback Actually, did the same for #21900 as well - e.g. replace `pd.` with direct imports

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @h-vetinari

**Pull request merged**